railties/test/application/configuration_test.rb

```diff
@@ -2654,6 +2654,28 @@ class ::DummySerializer < ActiveJob::Serializers::ObjectSerializer; end
       assert_includes ActiveJob::Serializers.serializers, DummySerializer
     end
 
+    test "use_big_decimal_serializer is disabled by default" do
+      app "test"
+
+      assert_not ActiveJob.use_big_decimal_serializer, "use_big_decimal_serializer should be disabled by default"
+    end
+
+    test "use_big_decimal_serializer can be enabled" do
+      app_file "config/environments/test.rb", <<-RUBY
+        Rails.application.configure do
+          Rails.application.config.active_job.use_big_decimal_serializer = true
+        end
+      RUBY
+
+      app "test"
+
+      # When loaded, ActiveJob::Base triggers the :active_job load hooks, which is where config is attached.
+      # Referencing the constant auto-loads it.
+      ActiveJob::Base
```

Comment:
This took a **long** time to figure out, as the assertion would just fail without it. It was particularly hard to test because the tests run in parallel with their output interleaved, and this suite would crash due to running out of file descriptors... (See #45667 😅)

It feels kind of weird that we have to do this. Nothing in the implementation `BigDecimalSerializer`, its use in `Assertions`, or let alone the configuration code references `ActiveJob::Base`, so it's weird that we have to reference it to get this to work. Is there a better way to do this, or is this acceptable?

I wonder if maybe I should have just set the config on `ActiveJob::Base` instead? That just felt like it attached it to a job, which `Arguments` isn't. 🤔

Reply:
It probably shouldn't be set on `on_load`, Active Record does it on `after_initialize`, might as well be consistent: https://github.com/rails/rails/blob/ab325a920986187f9d1c13bdfba77682ab6f6503/activerecord/lib/active_record/railtie.rb#L226-L238

Reply:
Okay, I've mirrored the approach, including the "double copying" for consistency. `ActiveRecord` does it for backwards compatibility though, which we don't technically need to support here (since `use_big_decimal_serializer` is the first config on `ActiveJob` directly).